core/src/main/java/nu/yona/server/rest/RestClientErrorHandler.java

```diff
@@ -0,0 +1,32 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Stichting Yona Foundation
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *******************************************************************************/
+package nu.yona.server.rest;
+
+import java.io.IOException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.client.ClientHttpResponse;
+import org.springframework.web.client.ResponseErrorHandler;
+
+public class RestClientErrorHandler implements ResponseErrorHandler
```

Comment:
Good that you point at this. Initially, it was only there to allow the callers of the REST template to handle the errors themselves. That lead to duplicate code, as every caller needed to do the same. Now I've moved the error handling code into this handler.